src/decorators/registry.ts

```diff
@@ -0,0 +1,26 @@
+import { RegistryLabel } from './../interfaces';
+import { WidgetRegistryItem } from './../WidgetRegistry';
+import { handleDecorator } from './../WidgetBase';
+
+/**
+ * The shape of the decorator configuration.
+ */
+export interface DefineDecoratorConfig {
+	label: RegistryLabel;
+	item: WidgetRegistryItem;
+}
+
+/**
+ * Decorator function to registry items that will be defined against
+ * the default registry of a widget.
+ *
+ * @param label The label of the registry item
+ * @param item The registry item to define
+ */
+export function define(label: RegistryLabel, item: WidgetRegistryItem) {
```

Comment:
again, more discussion, but `.add` would align it better to other _collections_ from an API consistency perspective... of course a more semantic name could be `register` as well... No strong feelings, but just poking at things to see if they stand up.

Reply:
The registry's API is `define` (as is custom elements) so it is aligned to that.

Reply:
Ah, you trumped me on API alignment... 🙇